doc/plutip-testing.md

```diff
@@ -26,13 +27,43 @@ CTL depends on a number of binaries in the `$PATH` to execute Plutip tests:
+   * and `ctl-scaffold-plutip-test` with the name of the plutip test package for your project;
+   * note that compilation of your project via Nix will fail in case there are any warnings, so you'd need to either fix or temporarily disable them.
 
 ## Testing contracts
 
-There are two entry points to the testing interface: `Contract.Test.Plutip.runPlutipContract` and `Contract.Test.Plutip.testPlutipContracts`. They work similarly, the difference being that `runPlutipContract` accepts a single `Contract` and runs in `Aff`, whereas `testPlutipContracts` interprets a `MoteT` (test tree of `ContractTest`s) into `Aff`. [Mote](https://github.com/garyb/purescript-mote) is a DSL for defining tests, and combined with `testPlutipContracts` you can use a single plutip instance to run multiple indepedent tests.
+CTL can help you test the offchain `Contract`s from your project (and consequently the interaction of onchain and offchain code) by spinning up a disposable private testnet via Plutip and making all your `Contract`s interact with it.
+
+There are two approaches to writing such tests.
+
+[First](#testing-in-aff-context) is to use the `Contract.Test.Plutip.runPlutipContract` function, which takes a single `Contract`, launches a Plutip cluster and executes the passed contract.
+This function runs in `Aff`; it will also throw an exception should contract fail for any reason.
+After the contract execution the Plutip cluster is terminated.
+You can either call it directly from your test's main or use any library for grouping and describing tests which support effects in the test body, like Mote.
+
+[Mote](https://github.com/garyb/purescript-mote) is a DSL for defining and grouping tests (plus other quality of life features, e.g. skipping marked tests).
+
+
+[Second](#testing-with-mote) (and more widely used) approach is to first build a tree of tests (in CTL's case a tree of `ContractTest` types -- basically a function from some distribution of funds to a `Contract a`) via Mote and then use the `Contract.Test.Plutip.testPlutipContracts` function to execute them.
+This allows to set up a Plutip cluster only once per top-level groups and tests passed to the `testPlutipContracts` and then use it in many independent tests.
+The function will interpret a `MoteT` (effectful test tree) into `Aff`, which you can then actually run.
+
+The [`ctl-scaffold` template](../templates/ctl-scaffold) provides a simple `Mote`-based example.
+
+
+CTL will run contracts in your test bodies and will print errors for any failed tests.
+Only test body failures are checked and this works fine if you want to make sure your `Contract`s execute without errors; if you want to add more precise checks (like checking that particular token is now at some address, that some exact amount was transferred, etc.) then you need to either write these checks manually in a `Contract` monad and then throw errors, or (preferably) use the [assertions library](./test-utils.md).
```

Comment:
> either

It's not either. It's either or both. Re-phrase it starting with "there are two options available..."

Reply:
Fixed